NetSSL_OpenSSL/src/HTTPSClientSession.cpp

```diff
@@ -117,6 +119,22 @@ void HTTPSClientSession::abort()
 }
 
 
+StreamSocket HTTPSClientSession::detachSocket()
+{
+	Poco::Net::SecureStreamSocket *oldSocket = dynamic_cast<Poco::Net::SecureStreamSocket *>(&socket());
+	if (oldSocket)
+	{
+		Poco::Net::SecureStreamSocket newSocket(oldSocket->context());
+		attachSocket(newSocket);
+		return *oldSocket;
```

Comment:
This is suspicious: what happens to the old socket object in `attachSocket()`? Does it get destroyed? If yes, then you implicitly construct a `StreamSocket` from `*oldSocket` which has been destroyed. This will cause undefined behavior.

Reply:
HTTPSession (parent class) has private mamber `_socket`, `socket()` returns it, `attachSocket(new_socket)` does `_socket = new_socket`. So `oldSocket` gets destroyed, but we must return it (so my previous comment might have been misleading, we want to get Context from it and then return it).

If we want to return it should we use `SecureStreamSocket oldSocket = static_cast<Poco::Net::SecureStreamSocket>(socket())` or something similar with `dynamic_cast`?

Reply:
https://pocoproject.org/docs/Poco.Net.HTTPSession.html#24153
```c++
StreamSocket & socket();
```
It returns a **reference** to the socket. Then you get a pointer to the same object cast to a `SecureStreamSocket` in `oldSocket`. Then you call `detachSocket()` -- this destroys the object pointed by `oldSocket`. Then you use `oldSocket` in the return statement, but it does not point to a valid object anymore => undefined behavior!!